simplematrixbotlib/bot.py

```diff
@@ -39,7 +39,7 @@ async def main(self):
 
 
         resp = await self.async_client.sync(timeout=65536,
-                                     full_state=False)  #Ignore prior messages
+                                     full_state=True)  #Ignore prior messages
```

Comment:
seems to be required to reliably load `self.room.own_user_id` and `self.room.users` which may be empty otherwise from testing. I hope there is a better way to do it than just syncing everything.

Reply:
Does self.room.own_user_id follow the structure of @<username>:<homeserver> ? If so, it would not be neccesary to do anything with self.room.users to obtain it.

Reply:
`self.room` is a `Dict[str, MatrixUser]`. `MatrixUser` contains `display_name` and `disambiguated_name` which we need for `mention()` matches

Reply:
maybe this https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.rooms.MatrixRoom.user_name is good enough instead?

Reply:
If we need the user_id, then whoami should solve that.

Reply:
Lines 57-60 of api.py
```
            async with aiohttp.ClientSession() as session:
                async with session.get(f'{self.creds.homeserver}/_matrix/client/r0/account/whoami?access_token={self.creds.access_token}') as response:
                    device_id = ast.literal_eval((await response.text()).replace(":false,", ":\"false\","))['device_id']
                    user_id = ast.literal_eval((await response.text()).replace(":false,", ":\"false\","))['user_id']
```

Reply:
Would the user id not be stored in self.async_client.user_id ?

Reply:
it would. the issue is more about getting the displayname though

Reply:
I think doing a full sync is actually ok if we enable storage in this PR (store is needed for #79)

https://github.com/poljar/matrix-nio/blob/a4fb83fd515568e269646d2111dc68e17cc251c6/nio/client/async_client.py#L1919

then only the very first time would be a "big" sync

Reply:
That is acceptable.